WooCommerce/Classes/ViewRelated/Coupons/Add and Edit Coupons/AddEditCoupon.swift

```diff
@@ -1,9 +1,92 @@
 import SwiftUI
 import Yosemite
 
+final class AddEditCouponHostingController: UIHostingController<AddEditCoupon> {
+
+    private let viewModel: AddEditCouponViewModel
+
+    init(viewModel: AddEditCouponViewModel) {
+        self.viewModel = viewModel
+        super.init(rootView: AddEditCoupon(viewModel))
+
+        // Needed because a `SwiftUI` cannot be dismissed when being presented by a UIHostingController
+        rootView.dismissHandler = { [weak self] in
+            self?.dismiss(animated: true)
+        }
+    }
+
+    required dynamic init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        handleSwipeBackGesture()
```

Comment:
We will not need this since the `AddEditCoupon` view is presented modally. We only need to check for the swipe back gesture if it's pushed into a navigation stack (using `navigationController.push()`).

Instead, we should intercept the swipe gesture when the view is presented by setting a delegate to its presentation controller here:

```suggestion
        presentationController?.delegate = self
```

Since you have implemented the conformance to `UIAdaptivePresentationControllerDelegate` below, this should be sufficient for detecting the swipe gesture.

Reply:
I can't believe I forgot to register the delegate 🤦, this was the missing part to make attempt 3 work. Thank you so much for catching this 🙇🙇🙇 It's fixed now!